### ISSUE TITLE
Do not check if Port is already in use with --validate-config

### ISSUE THREAD

**Issue by mnietz** [patroni/patroni#3056]
### What happened?

It's not a Bug but a Feature-Request.

When validating the Patroni-Config on a running cluster it exit with code 1 and the following messages: 

```
$ patroni --validate-config /etc/patroni/patroni.yml ; echo $?
restapi.listen 0.0.0.0:8008 didn't pass validation: Port 8008 is already in use.
postgresql.listen 0.0.0.0:5432 didn't pass validation: Port 5432 is already in use.
1
```

This makes it not possible to validate the config prior to a restart, for example within an ansible-template using validate:

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/template_module.html#parameter-validate

It would be pretty good to have a check/validate mode like nginx -t or haproxy -c that could be integrated into automation-tools.

### How can we reproduce it (as minimally and precisely as possible)?

Just validate the config of an already running cluster

### What did you expect to happen?

exit with 0 if config is valid

### Patroni/PostgreSQL/DCS version

-   Patroni version: 3.3
-   PostgreSQL version: 17 
-   DCS (and its version): etcd 3.5.7

### Patroni configuration file

```yaml
n/a
```

### patronictl show-config

```yaml
n/a
```

### Patroni log files

```shell
n/a
```

### PostgreSQL log files

```shell
n/a
```

### Have you tried to use GitHub issue search?

- [X] Yes

**Comment by sahilnaphade** [comment]
I would like to work on this.

I would prefer to have a different flag that would verify everything except port-in-use checks. Maybe something like `--validate-skip-port`?
Alternatively we can have a suppressor flag that can suppress the port-in-use failure.

@mnietz Would you say the first option satisfies the requirement?
CC: @CyberDem0n

**Comment by CyberDem0n** [comment]
@sahilnaphade sorry for a late reply.
I think we can just ignore `listen` parameter in the validate_host_port() function.
Creating a separate argument to modify behavior could be tricky, because of the way how schema validator is implemented.

**Comment by sahilnaphade** [comment]
@CyberDem0n If I understand correctly, are we considering ignoring the `listen` parameter for all schema validation cases?

If that's the case, I foresee a potential issue: if a port is already in use by a non-Patroni process, and the user tries to use that same port with a new configuration, it could lead to conflicts that won't be caught during validation.

To address this, I was thinking of introducing a **kwargs dictionary that could be passed down through the schema validation recursion. This way, we could maintain flexibility in how we handle specific validation scenarios, including the ability to selectively enforce the listen parameter when necessary.

**Comment by CyberDem0n** [comment]
@sahilnaphade Passing down parameters to Schema.validate() could be tricky.
Maybe we can create a global variable in the validator.py and change it based on additional parameters passed with `patroni --validate-config`, either directly, or by using a wrapper function.

What should be default behavior (to check listen or not) is a good question. Now I think we better stick by default with old behavior for backward compatibility.

**Comment by sahilnaphade** [comment]
@CyberDem0n Sounds good. I have raised the PR for this issue.
I have used the global variable approach, which, in my opinion would be easily extendable to any other use case we might have.

Please review and let me know if you want any further changes made: https://github.com/patroni/patroni/pull/3138

**Comment by CyberDem0n** [comment]
Solved in #3138
